### ISSUE TITLE
[BUG]:  Why is # 1 displayed instead of x1, x2

### ISSUE THREAD

**Issue by dadahuu** [MilesCranmer/PySR#1037]
### What happened?

Why is # 1 displayed instead of x1, x2？

`import numpy as np

X = 2 * np.random.randn(100, 2)
y = 2.5382 * np.cos(X[:, 0]) + X[:, 1] ** 2 - 0.5
from pysr import PySRRegressor
from pysr import TemplateExpressionSpec

template = TemplateExpressionSpec(
    combine        = "C[1] * cos(f(x1)) + g(x2) ^ 2 + C[2]",
    expressions    = ["f", "g"],
    variable_names = ["x1", "x2"],
    parameters     = {"C": 2},
)

model = PySRRegressor(
    maxsize          = 20,
    niterations      = 40,           # < Increase me for better results
    binary_operators = ["+", "*", "-", "/"],
    unary_operators  = [
        "cos",
        "exp",
        "sin",
        # "inv(x) = 1/x",
    ],
    # extra_sympy_mappings = {"inv": lambda x: 1 / x},
    expression_spec      = template,
    elementwise_loss     = "loss(prediction, target) = (prediction - target)^2",
)
model.fit(X, y)
# print(model)`

### Version

1.5.9

### Operating System

Windows

### Package Manager

pip

### Interface

Jupyter Notebook

### Relevant log output

```shell
Complexity  Loss       Score      Equation
2           2.464e+00  0.000e+00  ╭ f = #1
                                  ├ g = #1
                                  ╰ C = [0.34554, 0.14615]
4           1.727e-13  1.514e+01  ╭ f = #1 - 3.7543e-08
                                  ├ g = #1
                                  ╰ C = [2.5382, -0.5]
```
